File: dart_cli/test/sidecar_integration_test.dart
import 'dart:io';
import 'package:test/test.dart';
import 'package:dart_cli/src/sidecar.dart';

void main() {
  test('version check via sidecar', () async {
    final process = await Process.start(
        'cargo',
        [
          'run',
          '-p',
          'sidecar',
        ],
        workingDirectory: '../aider-core');
    addTearDown(() => process.kill());
    await Future.delayed(const Duration(seconds: 60));
    final info = await versionCheck();
    expect(info['current'], isNotEmpty);
  }, timeout: const Timeout(Duration(minutes: 2)));
}

File: go-shell/sidecarclient/sidecar_integration_test.go
package sidecarclient

import (
	"context"
	"os/exec"
	"testing"
	"time"
)

func TestVersionCheck(t *testing.T) {
	cmd := exec.Command("cargo", "run", "-p", "sidecar")
	cmd.Dir = "../../aider-core"
	if err := cmd.Start(); err != nil {
		t.Fatalf("failed to start sidecar: %v", err)
	}
	defer cmd.Process.Kill()
	time.Sleep(30 * time.Second)

	c := New()
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()
	info, err := c.VersionCheck(ctx)
	if err != nil {
		t.Fatalf("VersionCheck failed: %v", err)
	}
	if info.Current == "" {
		t.Fatalf("expected current version, got empty")
	}
}
